components/omega/src/ocn/PGrad.h

```diff
@@ -0,0 +1,178 @@
+#ifndef OMEGA_PGRAD_H
+#define OMEGA_PGRAD_H
+//===-- ocn/PGrad.h - Pressure Gradient -----------------*- C++ -*-===//
+///
+/// Implements the PressureGrad class which provides a centered and
+/// high-order pressure gradient option and dispatches computations to
+/// functor objects. This follows the patterns used in Eos.h/Eos.cpp.
+//
+//===----------------------------------------------------------------------===//
+
+#include "Config.h"
+#include "Eos.h"
+#include "HorzMesh.h"
+#include "OceanState.h"
+#include "OmegaKokkos.h"
+#include "VertCoord.h"
+#include <memory>
+
+namespace OMEGA {
+
+enum class PressureGradType { Centered, HighOrder1, HighOrder2 };
+
+/// Centered pressure gradient functor
+class PressureGradCentered {
+ public:
+   bool Enabled;
+
+   /// constructor declaration
+   PressureGradCentered(const HorzMesh *Mesh);
+
+   // Compute centered pressure gradient contribution for given edge and
+   // vertical chunk. This appends results into the Tend array (in-place).
+   KOKKOS_FUNCTION void operator()(const Array2DReal &Tend, I4 IEdge, I4 KChunk,
+                                   const Array2DReal &PressureMid,
+                                   const Array2DReal &PressureInterface,
+                                   const Array2DReal &Geopotential,
+                                   const Array2DReal &LayerThick,
+                                   const Array2DReal &ZInterface,
+                                   const Array2DReal &SpecVol,
+                                   const Array2DReal &SpecVolInterface) const {
+      const I4 KStart = KChunk * VecLength;
+
+      const I4 ICell0      = CellsOnEdge(IEdge, 0);
+      const I4 ICell1      = CellsOnEdge(IEdge, 1);
+      const Real InvDcEdge = 1.0_Real / DcEdge(IEdge);
+
+      for (int KVec = 0; KVec < VecLength; ++KVec) {
+         const I4 K = KStart + KVec;
+
+         // Average quantities to edge
+         const Real PAlphaEdgeK =
+             0.5_Real *
+             (PressureInterface(ICell1, K) * SpecVolInterface(ICell1, K) +
+              PressureInterface(ICell0, K) * SpecVolInterface(ICell0, K));
```

Comment:
Let's chat about it but I think we want the `PAlpha` product to be averaged both to cell centers and to layer interfaces.